Dockerfile-Client

```diff
@@ -29,12 +29,12 @@ ENV PATH "${HOME}/.local/bin:${PATH}"
 COPY --from=builder /code/prereq.txt .
 RUN pip install --no-deps -r prereq.txt --no-cache-dir
 
-COPY requirements.txt .
+COPY requirements/requirements.txt .
 RUN pip install -r requirements.txt --no-cache-dir
 
 # Install gordo, packaged from earlier 'python setup.py sdist'
 COPY --from=builder /code/dist/gordo-packed.tar.gz .
-RUN pip install gordo-packed.tar.gz
+RUN pip install gordo-packed.tar.gz[full]
```

Comment:
Note that this does not use pinned versions :-/ So I think we should pip install the `txt` versions of mlflow / postgres. But then it gets quite important that those files are compatible with `requirement.txt`

Reply:
So I'm having doubts about this whole system of requirements, especially when introducing `extas_require` and now a directory dedicated to holding all these files. I've looked around at various projects, like Flask, and [their setup.py](https://github.com/pallets/flask/blob/master/setup.py) is seemingly reflective of the larger practice of not having such abundant and strict requirements. 

In this case, saying we need to install the hard pinned requirements essentially says that our project is so unstable it can't be trusted to be installed from flexible requirements. What do you think?

Reply:
From a science standpoint, I think the pinned deps are a good thing for reproducibility, which isn't a problem Flask has. If they aren't pinned,  I think it may make sense to include the deps in the metadata somehow, but then that would affect the serializer... hmmm.

Reply:
[sklearn](https://github.com/scikit-learn/scikit-learn/blob/master/setup.py), [pytorch](https://github.com/pytorch/pytorch/blob/master/setup.py), and [mxnet](https://github.com/deepinsight/mxnet/blob/master/python/setup.py) also don't use pinned versions for installs, from the science standpoint as you suggest.

Reply:
Flasks setup.py is solving a different problem than our pinned requirement files. There is a difference between listing the requirements needed to install a library/application, vs specifying a deploy environment. That is the point of the pinned requirement files. Think of them more as belonging to the dockerfile than belonging to the setup.py. If we had the dockerfiles in a seperate repository (I am not saying we should do that, but if we did) then I think we would have put the pinned requirement files in there instead of here.

Reply:
But miles, we are also not using pinned versions in our install_requires...